### PR TITLE
Host permissions

### PR DESCRIPTION
**Pull request by szymonzmyslony**

### CODE REVIEW

**Comment by lewistaylor** [comment]
@jonwoodburn the branch has conflicts with development, can you please try to resolve by merging development into the branch and fixing issues?

File: formhero/forms/permission_classes.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from rest_framework.permissions import BasePermission
+from formhero.apps.models import App
+
+
+class HostPermission(BasePermission):
+    def has_permission(self, request, view):
+        hosts = App.objects.values_list('host_list').get(pk=request.data['app'])[0]
+        thishost = request.get_host()
```
Comment (vekerdyb) [inline formhero/forms/permission_classes.py]: @szymonzmyslony Use underscored variable names.
`this_host` is easier to read, and easier to spot if it has been misspelled.

File: formhero/forms/tests.py [hunk 2]
```diff
@@ -1,4 +1,32 @@
...
-# Create your tests here.
+User = get_user_model()
+
+
+class RetrievePostFormTests(APITestCase):
+
+    def test_post_a_new_form(self):
+        app1 = App.objects.create(host_list=["testserver"])
+        handler = 'db'
+        data = {"app": app1.pk, "name": "Szymon", "handler": handler, "config": "test"}
```
Comment (vekerdyb) [inline formhero/forms/tests.py]: @szymonzmyslony Putting the dictionary values on separate lines makes the code more maintainable and readable.
https://dangerfarms.gitbooks.io/newcomers-guide/content/section-2/code_quality.html#python-specific-rules

File: formhero/forms/tests.py [hunk 3]
```diff
@@ -1,4 +1,32 @@
...
+from rest_framework import status
 
 
-# Create your tests here.
+User = get_user_model()
+
+
+class RetrievePostFormTests(APITestCase):
+
+    def test_post_a_new_form(self):
```
Comment (vekerdyb) [inline formhero/forms/tests.py]: @szymonzmyslony This should be called `test_should_create_new_form_when_valid_data_is_posted`
From the Newcomers Guide:

> `To aid naming tests, our convention is to start each test with should_, resulting in test names matching the pattern test_should_<action>_<when/if>_<condition>`

https://dangerfarms.gitbooks.io/newcomers-guide/content/section-4/django/

File: formhero/forms/tests.py [hunk 4]
```diff
@@ -1,4 +1,32 @@
...
+        data = {"app": app1.pk, "name": "Szymon", "handler": handler, "config": "test"}
+        url = reverse(CreateForm.URL_NAME, kwargs={'pk': app1.pk})
+        response = self.client.post(url, data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.data['app'], app1.pk)
+        self.assertEqual(response.data['name'], "Szymon")
+        self.assertEqual(response.data['handler'], handler)
+        self.assertEqual(response.data['config'], "test")
+
+    def test_should_not_allow_to_post(self):
```
Comment (vekerdyb) [inline formhero/forms/tests.py]: @szymonzmyslony Similarly to the above, test should be called `test_should_not_allow_to_create_new_form_when_invalid_data_is_posted`

File: formhero/providers/dbform/tests.py [hunk 5]
```diff
@@ -40,3 +43,4 @@ def test_should_check_form_submission_data_is_the_same_as_form_data(self):
         self.assertEqual(my_form_submission.form_data['message'], self.data['message'])
         self.assertEqual(my_form_submission.form_data['name'], self.data['name'])
         self.assertEqual(my_form_submission.form_data['inquirer email'], self.data['inquirer email'])
+>>>>>>> b9cc2e78cf519f2f1654d8eaa3c034d97897d0f0
```
Comment (vekerdyb) [inline formhero/providers/dbform/tests.py]: @szymonzmyslony There are merge artefacts in this file. Please fix it.

File: project/urls.py [hunk 6]
```diff
@@ -17,9 +17,16 @@
...
+]
 
 if settings.DEBUG:
     urlpatterns += [
         url(r'^api-auth/', include('rest_framework.urls', namespace='rest_framework')),
     ]
+
+
+
+
```
Comment (vekerdyb) [inline project/urls.py]: @szymonzmyslony There are too many empty lines here.
This should have been picked up by the commit hooks.

File: formhero/apps/models.py [hunk 7]
```diff
@@ -1,5 +1,7 @@
 from django.db import models
+from django.contrib.postgres.fields import ArrayField
 
 
 class App(models.Model):
     name = models.TextField()
+    host_list = ArrayField(models.CharField(max_length=100), blank=True, default=[])
```
Comment (lewistaylor) [inline formhero/apps/models.py:7]: @szymonzmyslony a more descriptive name would be `allowed_hosts`

**Pull request merged**